packages/eslint-plugin/src/rules/no-misused-promises.ts

```diff
@@ -486,13 +486,40 @@ function isFunctionParam(
   return false;
 }
 
-// Get the positions of parameters which are void functions (and not also
+function checkThenableOrVoid(
```

Comment:
Nit: it's specifically checking an argument. This naming confused me for a little bit.

```suggestion
function checkThenableOrVoidArgument(
```

An alternative would be to move this function declaration to _inside_ the function `voidFunctionArguments` function. That way it wouldn't need to take in the two `Set`s as parameters. I don't have a strong preference - up to you!

Reply:
I've renamed the function